### PR TITLE
ktm-53: Add react-router-dom in project

### LINKED ISSUE #53: Add react-router-dom in project
https://reactrouter.com/en/main

### PR DESCRIPTION
**Pull request by maxmotsyk**

### CODE REVIEW

File: client/src/main.tsx
Comment (kuslosolodka) [inline client/src/main.tsx:1]: i think we should add router in main file

File: client/src/pages/Layout.tsx
Comment (kuslosolodka) [inline client/src/pages/Layout.tsx:1]: i'm not sure if we need this for now

File: client/src/router.jsx
Comment (kuslosolodka) [inline client/src/router.jsx:1]: why jsx?

File: client/src/pages/Error.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const Error = (): JSX.Element => {
```
Comment (kuslosolodka) [inline client/src/pages/Error.tsx]: this page is suggesting to be a component, as Not Found page, i think

```suggestion
const Error = (): React.FC => {
```

File: client/src/pages/index.js
Comment (kuslosolodka) [inline client/src/pages/index.js:1]: why js?

File: client/src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,9 @@
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-function App() {
-  return <div>Hello world</div>;
-}
+import { RouterProvider } from 'react-router-dom';
+
+import { router } from './router.jsx';
+
+const App = (): JSX.Element => <RouterProvider router={router} />;
```
Comment (kuslosolodka) [inline client/src/App.tsx]: This is our main page - component also. I think we should move router in main file

```suggestion
const App = (): React.FC => <something mock>;
```

File: client/src/router.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+
+const router = createBrowserRouter([
+  {
+    path: '/',
+    element: <Layout/>,
+    errorElement: <Error />,
+    children: [
+      {
+        index: true,
```
Comment (kuslosolodka) [inline client/src/router.jsx]: Do we need this?
```suggestion
```

File: client/src/pages/index.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export { Error } from './Error.tsx';
```
Comment (kuslosolodka) [inline client/src/pages/index.js]:
```suggestion
```

**Review by kuslosolodka — CHANGES_REQUESTED** [review]
Add pls router component to src/libs/components/router folder

File: client/src/router.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import {
+  createBrowserRouter,
+} from 'react-router-dom';
+
+import {Error} from './pages';
```
Comment (kuslosolodka) [inline client/src/router.jsx:5]: Format this, pls. One space before, one space after

File: client/src/pages/Error.tsx (hunk 1 above)
Comment (kuslosolodka) [reply client/src/pages/Error.tsx]: We don't need this page for now. Let's do it later

File: client/src/router.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+const router = createBrowserRouter([
+  {
+    path: '/',
+    element: <Layout/>,
+    errorElement: <Error />,
+    children: [
+      {
+        index: true,
+        element: <><p>Hello World</p></>
```
Comment (kuslosolodka) [inline client/src/router.jsx]: Add app.tsx in this route

**Review by kuslosolodka — CHANGES_REQUESTED** [review]

File: client/src/router.jsx (hunk 5 above)
Comment (kuslosolodka) [inline client/src/router.jsx]:
```suggestion
import { Error } from './pages';
```

**Review by kuslosolodka — CHANGES_REQUESTED** [review]

File: client/src/router.jsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+import {Error} from './pages';
+import { Layout } from './pages/Layout';
+
+
+const router = createBrowserRouter([
+  {
+    path: '/',
+    element: <Layout/>,
+    errorElement: <Error />,
```
Comment (kuslosolodka) [inline client/src/router.jsx]:
```suggestion
```

**Review by kuslosolodka — CHANGES_REQUESTED** [review]

File: client/src/router.jsx (hunk 3 above)
Comment (maxmotsyk) [reply client/src/router.jsx]: yes, since this property specifies the main page when rendering

File: client/src/App.tsx
Comment (kuslosolodka) [inline client/src/App.tsx:1]: move App to pages folder

File: client/src/App.tsx [hunk 8]
```diff
@@ -1,7 +1,13 @@
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-function App() {
-  return <div>Hello world</div>;
-}
+import { Outlet } from 'react-router-dom';
+
+const App: React.FC<any> = () => {
```
Comment (kuslosolodka) [inline client/src/App.tsx]:
```suggestion
const App: React.FC = () => {
```

File: package.json [hunk 9]
```diff
@@ -52,6 +52,7 @@
     "typescript": "5.1.3"
   },
   "dependencies": {
+    "commit": "0.0.4",
```
Comment (kuslosolodka) [inline package.json]: npm i uninstall 
```suggestion
```

File: client/src/libs/components/router/router.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    children: [
+      {
+        index: true,
+        element: (
+          <>
+            <p>Hello World</p>
+          </>
+        ),
+      },
+    ],
```
Comment (kuslosolodka) [inline client/src/libs/components/router/router.tsx]: I think we don't need children for now

```suggestion
```

File: client/src/App.tsx [hunk 11]
```diff
@@ -1,7 +1,13 @@
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-function App() {
-  return <div>Hello world</div>;
-}
+import { Outlet } from 'react-router-dom';
```
Comment (kuslosolodka) [inline client/src/App.tsx]: I think we don't need outlet for now

```suggestion
```

File: client/src/App.tsx [hunk 12]
```diff
@@ -1,7 +1,13 @@
...
+
+const App: React.FC<any> = () => {
+  return (
+    <>
+      <Outlet />
+    </>
+  );
+};
```
Comment (kuslosolodka) [inline client/src/App.tsx]:
```suggestion
  return (
    <>
     <something mock>
    </>
  );
};
```

**Review by kuslosolodka — CHANGES_REQUESTED** [review]

File: client/src/pages/.gitkeep
Comment (kuslosolodka) [inline client/src/pages/.gitkeep:1]: i think we don't need that anymore

**Review by kuslosolodka — CHANGES_REQUESTED** [review]

**Review by kuslosolodka — APPROVED** [review]

File: client/src/router.jsx (hunk 6 above)
Comment (HIMDanny) [reply client/src/router.jsx]: Disagree with this solution. It is better to add RouterProvider into App, to have opportunity to extend App component in future

File: client/src/App.tsx
Comment (HIMDanny) [reply client/src/App.tsx:1]: Why it should be in the pages folder?

**Review by HIMDanny — APPROVED** [review]

**Pull request merged**